core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/upscale/StackUpscaleActions.java

```diff
@@ -214,6 +215,9 @@ protected void doExecute(StackScalingFlowContext context, CollectMetadataResult
                 if (InstanceGroupType.GATEWAY == ig.getInstanceGroupType()) {
                     Stack stack = stackService.getByIdWithListsInTransaction(context.getStack().getId());
                     InstanceMetaData gatewayMetaData = stack.getPrimaryGatewayInstance();
+                    if (null == gatewayMetaData && isRepair(variables)) {
+                        throw new CloudbreakServiceException("Could not get instance metadata from the cloud provider.");
```

Comment:
Can we indicate that the gateway metadata could not be got from the provider side?